readthedocs/subscriptions/signals.py

```diff
@@ -34,23 +34,43 @@ def remove_stripe_subscription(sender, instance, using, **kwargs):
 
 # pylint: disable=unused-argument
 @receiver(post_save, sender=Organization)
-def update_billing_information(sender, instance, created, **kwargs):
-    """Update billing email information."""
+def update_stripe_customer(sender, instance, created, **kwargs):
+    """Update email and metadata attached to the stripe customer."""
     if created:
         return
 
     organization = instance
     log.bind(organization_slug=organization.slug)
-    # pylint: disable=broad-except
-    try:
-        s_customer = stripe.Customer.retrieve(organization.stripe_id)
-        if s_customer.email != organization.email:
-            s_customer.email = organization.email
-            s_customer.save()
-    except stripe.error.StripeError:
-        log.exception('Unable to update the Organization billing email on Stripe.')
-    except Exception:
-        log.exception('Unknown error when updating Organization billing email on Stripe.')
+
+    stripe_customer = organization.stripe_customer
+    if not stripe_customer:
+        return
```

Comment:
We should log a warning here. I understand that all organization should have a stripe customer attached.